period.go

```diff
@@ -188,21 +186,17 @@ func (p Period[T]) PeriodicISODuration(duration string) (periodical[T], error) {
 	}), nil
 }
 
-func convertTime[T TimeZone](arg any) (Time[T], error) {
-	switch v := arg.(type) {
+func convertTime[T TimeZone, argType timeish[T]](argPtr unsafe.Pointer) (Time[T], error) {
+	var dummy argType
+	switch any(dummy).(type) {
 	case Time[T]:
-		return v, nil
+		return *(*Time[T])(argPtr), nil
 	case time.Time:
-		return In[T](v), nil
-	case string:
-		return ParseISO[T](v)
+		return In[T](*(*time.Time)(argPtr)), nil
 	case []byte:
-		return ParseISO[T](string(v))
+		return ParseISO[T](string(*(*[]byte)(argPtr)))
```

Comment:
Can we use this?
https://pkg.go.dev/unsafe#String

Reply:
Certainly it can. I've updated the code.